plugins/modules/inventory_intent.py

```diff
@@ -2732,6 +2777,88 @@ def get_diff_merged(self, config):
         if device_type == "FIREPOWER_MANAGEMENT_SYSTEM":
             config['http_port'] = self.config[0].get("http_port", "443")
 
+        config['ip_address_list'] = devices_to_add
+
+        if not config['ip_address_list']:
+            self.msg = "Devices '{0}' already present in Cisco Catalyst Center".format(self.have['devices_in_playbook'])
+            self.log(self.msg, "INFO")
+            self.result['changed'] = False
+            self.result['response'] = self.msg
+
+        # To add the devices in inventory
+        if config['ip_address_list']:
```

Comment:
line 2782 we check with not, and here we check with true.. any reason? is it if and else?

This is a very lengthy one. Can you optimize it? we can split into small APIs.